myia/testing/common.py

```diff
@@ -1,21 +1,45 @@
 """Common testing utilities."""
+from dataclasses import dataclass
+
 from ovld import ovld
 
+from myia import basics
 from myia.abstract import data
 from myia.abstract.to_abstract import precise_abstract
 from myia.ir.node import Constant, Graph
+from myia.testing import numpy_subset as np
 
 
 @precise_abstract.variant
 def _to_abstract(self, x: type):
     return data.AbstractAtom({"interface": x})
 
 
+@ovld
+def _to_abstract(self, x: str):  # noqa: F811
+    """Keep value for string."""
+    return data.AbstractAtom({"interface": str, "value": x})
+
+
 @ovld
 def _to_abstract(self, x: (data.GenericBase, data.AbstractValue)):  # noqa: F811
     return x
 
 
+@ovld
+def _to_abstract(self, x: list):  # noqa: F811
+    # Let's expect list to have same type for all list values.
+    if not x:
+        items = []
+    else:
+        assert len(x) == 1, x
+        (item,) = x
+        items = [
+            item if isinstance(item, data.AbstractValue) else _to_abstract(item)
```

Comment:
It would be better to overload `_to_abstract(item)` on AbstractValue to return the item.